contracts/strategies/convex/base/ConvexStrategy2TokenHodl.sol

```diff
@@ -0,0 +1,442 @@
+  bytes32 internal constant _DEPOSIT_ARRAY_POSITION_SLOT = 0xb7c50ef998211fff3420379d0bf5b8dfb0cee909d1b7d9e517f311c104675b09;
+  bytes32 internal constant _CURVE_DEPOSIT_SLOT = 0xb306bb7adebd5a22f5e4cdf1efa00bc5f62d4f5554ef9d62c1b16327cd3ab5f9;
+  bytes32 internal constant _HODLVAULT_SLOT = 0xc26d330f887c749cb38ae7c37873ff08ac4bba7aec9113c82d48a0cf6cc145f2;
+  bytes32 internal constant _POTPOOL_SLOT = 0x7f4b50847e7d7a4da6a6ea36bfb188c77e9f093697337eb9a876744f926dd014;
+  bytes32 internal constant _HODL_RATIO_SLOT = 0xb487e573671f10704ed229d25cf38dda6d287a35872859d096c0395110a0adb1;
+
+  uint256 public constant ratioBase = 10000;
+
+  address[] public WETH2deposit;
+  mapping (address => address[]) public reward2WETH;
+  mapping (address => bool) public useUni;
+  address[] public rewardTokens;
+
+  constructor() public BaseUpgradeableStrategy() {
+    assert(_POOLID_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.poolId")) - 1));
+    assert(_DEPOSIT_TOKEN_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.depositToken")) - 1));
+    assert(_DEPOSIT_RECEIPT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.depositReceipt")) - 1));
+    assert(_DEPOSIT_ARRAY_POSITION_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.depositArrayPosition")) - 1));
+    assert(_CURVE_DEPOSIT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.curveDeposit")) - 1));
+    assert(_HODLVAULT_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.hodlVault")) - 1));
+    assert(_POTPOOL_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.potPool")) - 1));
+    assert(_HODL_RATIO_SLOT == bytes32(uint256(keccak256("eip1967.strategyStorage.hodlRatio")) - 1));
+  }
+
+  function initializeBaseStrategy(
+    address _storage,
+    address _underlying,
+    address _vault,
+    address _rewardPool,
+    uint256 _poolID,
+    address _depositToken,
+    uint256 _depositArrayPosition,
+    address _curveDeposit,
+    address _hodlVault,
+    address _potPool,
+    uint256 _hodlRatio
+  ) public initializer {
+
+    BaseUpgradeableStrategy.initialize(
+      _storage,
+      _underlying,
+      _vault,
+      _rewardPool,
+      weth,
+      300, // profit sharing numerator
+      1000, // profit sharing denominator
+      true, // sell
+      0, // sell floor
+      12 hours // implementation change delay
+    );
+
+    address _lpt;
+    address _depositReceipt;
+    (_lpt,_depositReceipt,,,,) = IBooster(booster).poolInfo(_poolID);
+    require(_lpt == underlying(), "Pool Info does not match underlying");
+    require(_depositArrayPosition < 2, "Deposit array position out of bounds");
+    require(_hodlRatio <= 10000, "Needs to be smaller than 10000");
+    _setDepositArrayPosition(_depositArrayPosition);
+    _setPoolId(_poolID);
+    _setDepositToken(_depositToken);
+    _setDepositReceipt(_depositReceipt);
+    _setCurveDeposit(_curveDeposit);
+    _setHodlVault(_hodlVault);
+    _setPotPool(_potPool);
```

Comment:
Is the intention to use a PotPool of the corresponding vault?
If yes, we need to:
1. Add validation to make sure it's the right PotPool
2. Not all vaults use PotPool. Some still use NoMintRewardPool

Reply:
Right, I had not thought about the old `NoMintRewardPool` problem. I have seen your new versions go up where a part of the CVX belonging to PS is sent to an address to HODL. This is certainly the most practical solution. I was thinking we could expand on this, and I am not sure you want to do this fully, but what we could do is hodl all of the PS rewards, 30% of the CRV and 30% of the CVX (if there are additional rewards like LDO/BOR just notify them separately for 30% buyback). This CRV and CVX can be sent to the multisig and the multisig can stake those assets with Convex for rewards. CRV is currently getting 95% APR and CVX is getting 80% APR. We can then use these rewards to buyback FARM and reward PS. This allows us to be more synergetic with both Curve and Convex, and should earn good rewards for PS over time.

When launching new Curve vaults we can look at turning this around, as the `PotPools` give us more possibilities. We could put the strategy rewards for users into CRV and CVX staking vaults, while the PS part just buys back farm.